packages/menu/src/useMenu.ts

```diff
@@ -0,0 +1,659 @@
+
+        if (!isExpandedControlled || !isSelectionValueControlled) {
+          dispatch({
+            type: changeType,
+            payload: {
+              ...(isExpandedControlled ? {} : { isExpanded: false }),
+              ...(!isSelectionValueControlled && nextSelection
+                ? { selectedItems: nextSelection }
+                : {})
+            }
+          });
+        }
+
+        onChange({
+          type: changeType,
+          isExpanded: false,
+          ...(nextSelection ? { selectedItems: nextSelection } : {})
+        });
+      };
+
+      const handleItemKeyDown = (event: KeyboardEvent) => {
+        const { key } = event;
+        const isJumpKey = [KEYS.HOME, KEYS.END].includes(key);
+        const isSelectKey = [KEYS.SPACE, KEYS.ENTER].includes(key);
+        const isArrowKey = [KEYS.UP, KEYS.DOWN, KEYS.LEFT, KEYS.RIGHT].includes(key);
+        const isAlphanumericChar = key.length === 1 && /\S/u.test(key);
+
+        let changeType;
+        let payload;
+        let changes;
+
+        if (isSelectKey) {
+          event.preventDefault();
+
+          changeType = StateChangeTypes[toMenuItemKeyDownType(key)];
+          const nextSelection = getSelectionValue({ value, type, name, label, selected });
+
+          if (!isExpandedControlled || !isSelectionValueControlled) {
+            payload = {
+              ...(isExpandedControlled ? {} : { isExpanded: false }),
+              ...(!isSelectionValueControlled && nextSelection
+                ? { selectedItems: nextSelection }
+                : {})
+            };
+          }
+
+          changes = {
+            isExpanded: false,
+            ...(nextSelection ? { selectedItems: nextSelection } : {})
+          };
+        }
+
+        if (isArrowKey || isJumpKey || isAlphanumericChar) {
+          changeType = StateChangeTypes[toMenuItemKeyDownType(key)];
+          const nextFocusedValue = getNextFocusedValue(value, key, isAlphanumericChar);
+
+          if (isAlphanumericChar) {
+            changeType = isAlphanumericChar ? StateChangeTypes.MenuItemKeyDown : changeType;
+          }
+
+          payload = {
+            ...(isFocusedValueControlled ? {} : { focusedValue: nextFocusedValue })
+          };
+
+          changes = { focusedValue: nextFocusedValue };
+        }
+
+        if (changeType) {
+          payload && dispatch({ type: changeType, payload });
+
+          onChange({ type: changeType, ...(changes || {}) });
+        }
+      };
+
+      const handleItemMouseEnter = () => {
+        const changeType = StateChangeTypes.MenuItemMouseMove;
+
+        dispatch({
+          type: changeType,
+          payload: {
+            ...(isFocusedValueControlled ? {} : { focusedValue: value })
+          }
+        });
+
+        onChange({
+          type: changeType,
+          focusedValue: value
+        });
+      };
+
+      const itemProps = getElementProps({
+        value: value as any,
```

Comment:
Without this `any`, this error is given:

```
Type 'readonly string[] | ItemValue' is not assignable to type '(string | number | readonly string[] | undefined) & ItemValue'.
  Type 'readonly string[]' is not assignable to type '(string | number | readonly string[] | undefined) & ItemValue'.
    Type 'readonly string[]' is not assignable to type 'readonly string[] & number'.
      Type 'readonly string[]' is not assignable to type 'number'.
```

Seems like the issue is related to how the `value` parameter in `getElementProps` expects the native HTML `value` type. As a result, if `value` in `getItemProps` is top level, this error goes away, but since it's nested within the `item` parameter, it doesn't infer "correctly."

Separately, omitting `value` from `HTMLProps` in `IUseSelectionReturnValue['getElementProps']` doesn't resolve this. Having `getItemProps`'s `item.value` incorporate `HTMLProps['value']` is also no bueno.

Weird issue. Not sure what to do about it.

Reply:
I think the `any` fallback is acceptable in this case.